src/Engine/Entity/Entity.cpp

```diff
@@ -62,7 +62,7 @@ bool Entity::SetParent(Entity* newParent) {
 
 }
 
-bool Entity::HasChild(Entity* check_child, bool deep) {
+const bool Entity::HasChild(Entity* check_child, bool deep) {
```

Comment:
This const has no effect; bool is always copied anyway. Placing the const after the parameter list (before the brace) will indicate that the function itself is const.